codec/gcm.go

```diff
@@ -1,26 +1,58 @@
 package codec
 
 import (
+	"bytes"
 	"crypto/aes"
 	"crypto/cipher"
-	"encoding/binary"
 	"fmt"
 	"io"
-	"math/rand"
+	"math"
+
+	"github.com/renproject/id"
+	"github.com/renproject/surge"
 )
 
+type gcmNonce struct {
```

Comment:
Consider adding a comment to explain what `top` and `bottom` are (initially, this confused me a little).